packages/twenty-server/src/main.ts

```diff
@@ -17,17 +19,37 @@ import './instrument';
 
 import { settings } from './engine/constants/settings';
 import { generateFrontConfig } from './utils/generate-front-config';
+import { ServerUrl, ApiUrl } from './engine/utils/server-and-api-urls';
 
 const bootstrap = async () => {
   const app = await NestFactory.create<NestExpressApplication>(AppModule, {
     cors: true,
     bufferLogs: process.env.LOGGER_IS_BUFFER_ENABLED === 'true',
     rawBody: true,
     snapshot: process.env.DEBUG_MODE === 'true',
+    ...(process.env.SERVER_URL &&
+    process.env.SERVER_URL.startsWith('https') &&
+    process.env.SSL_KEY_PATH &&
+    process.env.SSL_CERT_PATH
+      ? {
+          httpsOptions: {
+            key: fs.readFileSync(process.env.SSL_KEY_PATH),
+            cert: fs.readFileSync(process.env.SSL_CERT_PATH),
+          },
+        }
+      : {}),
   });
   const logger = app.get(LoggerService);
   const environmentService = app.get(EnvironmentService);
 
+  const serverUrl = new URL(
+    environmentService.get('SERVER_URL').startsWith('http')
+      ? environmentService.get('SERVER_URL')
+      : `http://${environmentService.get('SERVER_URL')}`,
```

Comment:
It was to manage the case where a user gives a SERVER_URL with hostname without protocol like "localhost" instead of "http://localhost".

I will implement a better check.